### DISCUSSION TITLE
Network Tokenization

### DISCUSSION THREAD

**Discussion by euluizaraujo** [juspay/hyperswitch#3094] [Ideas / Feature Requests]
Hello, what is the forecast for the opensource launch of Network Tokenization?  Can we help with development in any way?  We work with subscriptions and this feature would help a lot.

**Comment by SanchithHegde** [comment]
Hey @euluizaraujo, network tokenization isn't on our roadmap for this quarter. I doubt we'll be able to pick it up anytime before Q2, 2024.

**Comment by gorakhnathy7** [comment]
Marking this as closed for now, will update once this is included in the roadmap.

**Comment by icelattefull** [comment]
@gorakhnathy7 according to the [roadmap](https://docs.hyperswitch.io/about-hyperswitch/roadmap-q3-2024/previous-roadmap-q3-2024), this was delivered in Q2 2024. Will this be made available in self hosted instances too?

**Comment by gorakhnathy7** [comment]
Hey @icelattefull Currently we don't support Network Tokenisation for Self Hosted instances, But if you can share more about your business use -case at biz@hyperswitch.io, we can probably help you better.

**Comment by cortopy** [comment]
There are two reasons why I think network tokenisation would be a nice thing to have in the self-hosted version:

- in the new pricing model of hyperswitch there's no free tier or pay-as-you-go options. This means that for small projects like mine self-hosting is the only option. Having network tokenisation would unblock saving card details in self-hosted instances without having to deploy the vault and PCI compliance checks.
- subscriptions: this greatly reduces cancellations, as it cover for cases like credit card expires

**Comment by gorakhnathy7** [comment]
Thanks for pointing these out, will let you know after discussing with the team.

**Comment by cortopy** [comment]
thanks @gorakhnathy7 , I can see a PR was merged to implement this feature in the latest [release](https://github.com/juspay/hyperswitch/releases/tag/v1.112.0). Does this mean this a feature now available in self hosted too?

**Comment by gorakhnathy7** [comment]
Hey @cortopy As of now, we don't have plans to roll out this feature for self hosted instances.
If you have a specific use case, kindly reach out to us via this [form](https://hyperswitch.io/contact-us) for us to help you better.
